### ISSUE TITLE
Wrong error position in the presence of implicit arguments

### ISSUE THREAD

**Issue by odersky** [scalameta/metals#2873]
**Describe the bug**

**To Reproduce**

Load the following program:

```scala
class Ctx

class Tree:
  def span(using Ctx): String = ???

def f(x: Int) = ???

def g(tree: Tree)(using Ctx) = f(tree.span)
```
You'll see a red underline under `tree` with error message:
```
Found: String
Required: Int
```
**Expected behavior**

The error should be on `span`, not on `tree`. If I drop the `(using Ctx)` clauses, the error is correctly diagnosed on `span`. 
By comparison, if I compile in the terminal, the underline goes over the whole selection.
```scala
8 |def g(tree: Tree)(using Ctx) = f(tree.span)
  |                                 ^^^^^^^^^
  |                                 Found:    String
  |                                 Required: Int
```

**Comment by dos65** [comment]
The actual reason is in `xsbit.Position`  implementation in scala3 compiler.
In order to have correct ranges in diagnostic, it has to override a bunch of additional methods: `startLine`,`startColumn`, `endColumn` ...

Fixed in: https://github.com/lampepfl/dotty/pull/12838

**Comment by dos65** [comment]
Fixed from 3.0.0-RC2
